common/lib/dependabot/notices.rb

```diff
@@ -0,0 +1,139 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+module Dependabot
+  class Notice
+    extend T::Sig
+
+    sig { returns(String) }
+    attr_reader :mode, :type
+
+    sig { returns(T.nilable(String)) }
+    attr_reader :package_manager_name, :message, :markdown
+
+    sig { returns(T::Hash[Symbol, T.untyped]) }
+    attr_reader :details
+
+    sig do
+      params(
+        mode: String,
+        type: String,
+        package_manager_name: T.nilable(String),
+        message: T.nilable(String),
+        markdown: T.nilable(String),
+        details: T::Hash[Symbol, T.untyped]
```

Comment:
I'm trying to understand why we have a `details` Hash? Message and Markdown are already instance variables that hold the same data.

Reply:
Let's keep this change focused on adding a warning to the PR body in the Updater and discuss how we'll add the warning in the UI next.

Reply:
removed details from the notice.